app/src/main/java/ch/epfl/skysync/viewmodel/LocationViewModel.kt

```diff
@@ -8,23 +9,30 @@ import androidx.lifecycle.viewmodel.compose.viewModel
 import ch.epfl.skysync.Repository
 import ch.epfl.skysync.components.SnackbarManager
 import ch.epfl.skysync.database.ListenerUpdate
-import ch.epfl.skysync.models.flight.ConfirmedFlight
+import ch.epfl.skysync.models.flight.Flight
 import ch.epfl.skysync.models.flight.RoleType
+import ch.epfl.skysync.models.flight.Team
 import ch.epfl.skysync.models.location.FlightTrace
 import ch.epfl.skysync.models.location.Location
 import ch.epfl.skysync.models.user.User
+import ch.epfl.skysync.util.WhileUiSubscribed
 import com.google.firebase.firestore.Filter
 import com.google.firebase.firestore.ListenerRegistration
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
 import kotlinx.coroutines.launch
 
-class LocationViewModel(val uid: String, repository: Repository) : ViewModel() {
+/** ViewModel for the location tracking of the user during a flight and the timer. */
+class LocationViewModel(uid: String? = null, val repository: Repository) : ViewModel() {

```

Comment:
Maybe rename the view model to something clearer as it does not handle only locations now

Reply:
I agree, but we'll do that in a separate PR since renaming classes combined with a lot of modifications does confuse  Git and I wanted that changes are clearly visible